src/gdb_main.c

```diff
@@ -618,14 +618,15 @@ static void handle_v_packet(char *packet, const size_t plen)
 				tok++;
 				continue;
 			}
-			if (isxdigit(tok[0]) && isxdigit(tok[1])) {
-				unhexify(pcmdline, tok, 2);
+			/* isxdigit expects int, to handle EOF */
+			if (isxdigit((int8_t)*tok) && isxdigit((int8_t) * (tok + 1U))) {
```

Comment:
This reads a bit confusingly after, especially with how clang-format formats this.. suggestion:
```suggestion
			if (isxdigit((int8_t)tok[0U]) && isxdigit((int8_t)tok[1U])) {
```